pkg/controller/resources/route.go

```diff
@@ -128,6 +124,14 @@ func makeRoute(ci networkingv1alpha1.IngressAccessor, host string, rule networki
 		return nil, ErrNoValidLoadbalancerDomain
 	}
 
+	secret := &corev1.Secret{}
+	if cert, ok := annotations[CertificateAnnotation]; ok {
+		err := client.Get(context.TODO(), types.NamespacedName{Namespace: ci.GetNamespace(), Name: cert}, secret)
```

Comment:
Can we consider to pass this secret into `MakeRoutes` from the outside? That'd keep this function lean and we could remove the error returns.

Reply:
Sure.
Or, we should support only default certs for now? If we have a plan to migrate knative-serving-networking-openshift or kourier, we can choose minimal support for TLS for now.

I have sent the PR against master branch https://github.com/openshift-knative/knative-openshift-ingress/pull/36 after dropping custom cert to compare it with this.